src/main/kotlin/com/google/androidstudiopoet/utils/DependencyImageGenerator.kt

```diff
@@ -0,0 +1,130 @@
+    }
+
+    /**
+     * generates an image that represents the dependency matrix of the project
+     */
+    fun generate(blueprint: ProjectBlueprint) {
+        val numModules = blueprint.moduleBlueprints.size + blueprint.androidModuleBlueprints.size
+        val imageSize = HEADER_SIZE + LINE_WIDTH + numModules * GRID_SIZE
+
+        // Create buffered image object
+        val img = BufferedImage(imageSize, imageSize, BufferedImage.TYPE_INT_RGB)
+        val graphics = img.graphics
+
+        // Background
+        graphics.color = BACKGROUND_COLOR
+        graphics.fillRect(0, 0, imageSize, imageSize)
+
+        // Generate the grid
+        graphics.color = GRID_COLOR
+        for (i in 0..numModules) {
+            // Horizontal
+            val y = HEADER_SIZE + i * GRID_SIZE
+            graphics.fillRect(HEADER_SIZE, y, imageSize - HEADER_SIZE, LINE_WIDTH)
+            // Vertical
+            val x = HEADER_SIZE + i * GRID_SIZE
+            graphics.fillRect(x, HEADER_SIZE, LINE_WIDTH, imageSize - HEADER_SIZE)
+        }
+
+        // Add Android modules headers
+        val numAndroid = blueprint.androidModuleBlueprints.size
+        for ((index, module) in blueprint.androidModuleBlueprints.withIndex()) {
+            if (module.hasLaunchActivity) {
+                graphics.drawHeader(index, APP_COLOR)
+            }
+            else {
+                graphics.drawHeader(index, ANDROID_COLOR)
+            }
+
+            // Dependencies
+            for (dependency in module.moduleDependencies) {
+                val split = DependencyValidator.ModuleSplit(dependency.name)
+                val dependencyIndex = if (split.type == "module") {
+                    // Java module
+                    numAndroid + split.index
+                } else {
+                    split.index
+                }
+                graphics.drawCell(index, dependencyIndex, DEPENDENCY_COLOR)
+            }
+        }
+
+        // Now Java modules
+        for ((index, module) in blueprint.moduleBlueprints.withIndex()) {
+            graphics.drawHeader(index + numAndroid, JAVA_COLOR)
+            // Dependencies
+            for (dependency in module.moduleDependencies) {
+                val split = DependencyValidator.ModuleSplit(dependency.name)
+                val dependencyIndex: Int
+                val dependencyColor: Color
+                if (split.type == "module") {
+                    // Java module
+                    dependencyIndex = numAndroid + split.index
+                    dependencyColor = DEPENDENCY_COLOR
+                } else {
+                    dependencyIndex = split.index
+                    dependencyColor = ERROR_COLOR
+                }
+                graphics.drawCell(index + numAndroid, dependencyIndex, dependencyColor)
+            }
+        }
+
+        // Save to file
+        val imgPath = blueprint.projectRoot.joinPath("dependencies.png")
+        ImageIO.write(img,"png", File(imgPath))
```

Comment:
Please make `ImageWriter` similar to `FileWriter`, and use it here and in `ImageGenerator`

Reply:
There are two writers now, with that, it is easier to create tests for DependencyGraphGeneration, thanks for suggesting it!